### PR TITLE
Fix build failures of Device Shadow and Jobs demos from FreeRTOS console

### PR DESCRIPTION
**Pull request by aggarw13**
The FreeRTOS console download with enabled **Device Shadow** and/or **Jobs** libraries fails. Demos of both libraries depend on `AFR::mqtt_demo_helpers` target which was missing in the `AFR::<device_shadow/jobs>_demo_dependencies` wrapper target for FreeRTOS console. Also, the path of `shadowFilePaths.cmake` was stale.

### CODE REVIEW

File: libraries/device_shadow_demo_dependencies.cmake [hunk 1]
```diff
@@ -17,7 +17,7 @@ endforeach()
 
 # Add cmake files of module to metadata.
 afr_module_cmake_files(${AFR_CURRENT_MODULE} 
-    ${CMAKE_CURRENT_LIST_DIR}/device_shadow_for_aws_iot_embedded_sdk/shadowFilePaths.cmake
```
Comment (leegeth) [inline libraries/device_shadow_demo_dependencies.cmake:20]: was this an issue in last release or caused by the merge of release?

**Review by leegeth — DISMISSED** [review]

**Review by leegeth — DISMISSED** [review]

File: libraries/device_shadow_demo_dependencies.cmake (hunk 1 above)
Comment (aggarw13) [reply libraries/device_shadow_demo_dependencies.cmake:20]: This is caused by the merger of the `release` branch into `master`

File: libraries/device_shadow_demo_dependencies.cmake [hunk 2]
```diff
@@ -88,3 +88,15 @@ afr_module_dependencies(
...
+# Device Shadow demo that uses Secure Sockets 
+# (present at demos/device_shadow_for_aws folder) ONLY if
+# the board supports the Secure Sockets library.
+if(TARGET AFR::secure_sockets::mcu_port)
+    afr_module_dependencies(
+        ${AFR_CURRENT_MODULE}
+        PUBLIC
+            AFR::mqtt_demo_helpers
+    )
+endif()
```
Comment (leegeth) [inline libraries/device_shadow_demo_dependencies.cmake:102]:
```suggestion
endif()

```

File: libraries/device_shadow_demo_dependencies.cmake [hunk 3]
```diff
@@ -82,8 +82,21 @@ afr_module_dependencies(
     PUBLIC
         AFR::device_shadow
         AFR::core_json
+        AFR::mqtt_demo_helpers
```
Comment (aggarw13) [inline libraries/device_shadow_demo_dependencies.cmake:85]:
```suggestion
```

**Review by leegeth — APPROVED** [review]

**Review by gkwicker — APPROVED** [review]

**Pull request merged**